internal/controller/controllers/infraenv_controller.go

```diff
@@ -93,6 +93,25 @@ type InfraEnvReconciler struct {
 // +kubebuilder:rbac:groups=agent-install.openshift.io,resources=nmstateconfigs,verbs=get;list;watch
 // +kubebuilder:rbac:groups=agent-install.openshift.io,resources=infraenvs,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=agent-install.openshift.io,resources=infraenvs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=agent-install.openshift.io,resources=agentserviceconfigs,verbs=get;list;watch
+
+// isImageServiceDisabled checks if the image service is disabled via AgentServiceConfig annotation
+func (r *InfraEnvReconciler) isImageServiceDisabled(ctx context.Context) bool {
```

Comment:
Same here, what do you think about just setting a bool in the controller that says whether or not image service is disabled instead of querying for the AgentServiceConfig in this controller?